core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala

```diff
@@ -807,29 +937,49 @@ object ConsumerGroupCommand extends Logging {
       if (options.has(timeoutMsOpt) && (!describeOptPresent || useOldConsumer))
-        debug(s"Option '$timeoutMsOpt' is applicable only when both '$bootstrapServerOpt' and '$describeOpt' are used.")
+        debug(s"Option $timeoutMsOpt is applicable only when both $bootstrapServerOpt and $describeOpt are used.")
 
       if (useOldConsumer) {
         if (options.has(bootstrapServerOpt))
-          CommandLineUtils.printUsageAndDie(parser, s"Option '$bootstrapServerOpt' is not valid with '$zkConnectOpt'.")
+          CommandLineUtils.printUsageAndDie(parser, s"Option $bootstrapServerOpt is not valid with $zkConnectOpt.")
         else if (options.has(newConsumerOpt))
-          CommandLineUtils.printUsageAndDie(parser, s"Option '$newConsumerOpt' is not valid with '$zkConnectOpt'.")
+          CommandLineUtils.printUsageAndDie(parser, s"Option $newConsumerOpt is not valid with $zkConnectOpt.")
       } else {
         CommandLineUtils.checkRequiredArgs(parser, options, bootstrapServerOpt)
 
         if (options.has(newConsumerOpt)) {
-          Console.err.println("The --new-consumer option is deprecated and will be removed in a future major release." +
-            "The new consumer is used by default if the --bootstrap-server option is provided.")
+          Console.err.println(s"The $newConsumerOpt option is deprecated and will be removed in a future major release." +
+            s"The new consumer is used by default if the $bootstrapServerOpt option is provided.")
         }
 
         if (options.has(deleteOpt))
-          CommandLineUtils.printUsageAndDie(parser, s"Option '$deleteOpt' is only valid with '$zkConnectOpt'. Note that " +
-            "there's no need to delete group metadata for the new consumer as the group is deleted when the last " +
+          CommandLineUtils.printUsageAndDie(parser, s"Option $deleteOpt is only valid with $zkConnectOpt. Note that " +
+            "there is no need to delete group metadata for the new consumer as the group is deleted when the last " +
             "committed offset for that group expires.")
       }
 
       if (describeOptPresent)
         CommandLineUtils.checkRequiredArgs(parser, options, groupOpt)
+
+      val membersOptPresent = options.has(membersOpt)
+      val offsetsOptPresent = options.has(offsetsOpt)
+      val stateOptPresent = options.has(stateOpt)
+      val subActions = Seq(membersOptPresent, offsetsOptPresent, stateOptPresent).count(_ == true)
+      if (subActions > 0) {
+        if (useOldConsumer)
+          CommandLineUtils.printUsageAndDie(parser, s"Options $membersOpt, $offsetsOpt, and $stateOpt may only be used " +
+              s"when $bootstrapServerOpt is present. They do not support ZooKeeper-based consumer group management.")
+
+        if (!describeOptPresent)
+          CommandLineUtils.printUsageAndDie(parser,
+              s"Options $membersOpt, $offsetsOpt, and $stateOpt may only be used with $describeOpt.")
+        if (subActions > 1)
```

Comment:
The options parser has a `mutuallyExclusive` function that might already provide this.

Reply:
Yeah, I assumed that if you added the `mutuallyExclusive` clause, then parsing itself would fail and we could leave this check out.

Reply:
Let me move it up before `parse`. It would still report the same mutually exclusive error for when old consumer is used. But perhaps it's okay. Let me know if you think otherwise. Thanks.